### ISSUE TITLE
Play Sprint Plan 2019-03-25

### ISSUE THREAD

**Issue by marcospereira** [playframework/play-meta#56]
[Previous sprint plans](https://github.com/lightbend/play-meta/issues?q=label:sprint-plan+sort:created-desc)

## Dates

- Start: 2019-03-25
- End: 2019-04-05

## Absences

- @marcospereira:
    - Support: April 1 – 5, 2019
- @ignasi35:
    - Support: March 25 – 29, 2019
- @dwijnand:
    - Off: March 29 – April 2, 2019

### Cleanups

* [x] Evaluate the impact of Lagom 1.5 on the Operations training ([Email thread](https://groups.google.com/a/lightbend.com/d/msg/play-team/d4-vOT-JDWA)) @renatocaval 
* [ ] Play: Release 2.7.1 https://github.com/playframework/play-meta/issues/52 @marcospereira 
* [x] Upgrade builds to sbt 1 https://github.com/playframework/play-meta/issues/14
    * [x] Play https://github.com/playframework/playframework/issues/8383 @marcospereira
	* [x] Play 2.6 https://github.com/playframework/playframework/pull/9140
    * [x] Lagom https://github.com/lagom/lagom/issues/1329 @dwijnand
    * [x] lagom.com https://github.com/lagom/lagom.github.io/pull/183 @ignasi35 
    * [x] omnidoc https://github.com/playframework/omnidoc/pull/32 @ignasi35
    * [x] play-slick https://github.com/playframework/play-slick/pull/465 @ignasi35 
    * [x] scalatestplus-play https://github.com/playframework/scalatestplus-play/pull/110 and https://github.com/playframework/scalatestplus-play/pull/145 @marcospereira @ignasi35 
* [x] (unplanned) Cleanup stable branches. Stable branches are not up to date with `needs-backport` PRs. 
    * [x] scalatestplus-play (needed backport and forward-ports)
    * [x] play-ws
    * [x] play-slick 
* [ ] Drop allow_failures for AdoptOpenJDK 11 (+ update spreadsheet + tracking issue) @dwijnand 
    * [x] https://github.com/playframework/cachecontrol/pull/49 
	* [x] https://github.com/playframework/play-file-watch/pull/35
	* [x] https://github.com/playframework/twirl/pull/208
	* [x] https://github.com/playframework/netty-reactive-streams/pull/54
	* [x] https://github.com/lagom/lagom-akka-discovery-service-locator/pull/63 
	* [x] https://github.com/playframework/scalatestplus-play/pull/140
	* [x] play
	    * [x] Run scripted tests in AdoptOpenJDK 11 (sbt 1) https://github.com/playframework/playframework/pull/9133
		* [x] Update Akka HTTP to version 10.1.8 https://github.com/playframework/playframework/pull/9116
	* [x] ~play-iteratees~
	* [x] play-slick https://github.com/playframework/play-slick/pull/464
    * [x] play-json https://github.com/playframework/play-json/pull/244
    * [x] play-ws https://github.com/playframework/play-ws/pull/315
	* [ ] lagom https://github.com/lagom/lagom/issues/1417
 * [ ] play-grpc: cross-build for Scala 2.13 https://github.com/playframework/play-grpc/issues/40
    * [x] blocked on https://github.com/akka/akka-grpc/issues/506
    * [ ] blocked on https://github.com/lagom/lagom/issues/1240
		* [ ] blocked on [slick/slick#1958](https://github.com/slick/slick/issues/1958)
		* [ ] blocked on [akka/alpakka-kafka#540](https://github.com/akka/alpakka-kafka/issues/540)
		* [ ] blocked on [apache/kafka#5454](https://github.com/apache/kafka/pull/5454)
* [x] Remove sbteclipse from all the repos https://github.com/playframework/play-meta/issues/50 @ignasi35 
* [x] Set up org-wide GitHub files https://github.com/playframework/play-meta/issues/47
* [x] ~~Lagom: shopping-cart > hello world https://github.com/playframework/play-meta/issues/33~~ deferred until further discussed
    * [x] ~~Archive chirper~~
    * [x] ~~Archive online-auction~~
        * [x] ~~Add a Play web-gateway in shopping-cart~~
        * [x] ~~Cassandra example?~~
    * [x] ~~Is it possible to use shopping-cart to smoke test the deploy to OpenShift?~~
* [x] TemplateControl: handle Mergify changes https://github.com/playframework/templatecontrol/pull/113 @dwijnand
* [x] Use scala-steward: https://github.com/playframework/play-meta/issues/57 https://github.com/fthomas/scala-steward/pull/378
    * [x] ~Work together with Mergify~ - deferred. We can start by merging the PRs manually. 
* [x] Lagom cron build
    * [x] Fix the build https://github.com/lagom/lagom/pull/1823
    * [x] Build for 1.5.x branch
* [x] Play cron build is broken @renatocaval - PR: https://github.com/playframework/playframework/pull/9124
* [ ] g8 scaffolding
    * [ ] For Lagom services
    * [ ] For Play forms/controllers
* [ ] Mergify
    * [x] Do backports - REVERTED: we found issues with Mergify
          https://github.com/Mergifyio/mergify-engine/issues/359
    * [ ] Inspect /lagom/ and /playframework/ repositories to add Mergify: https://github.com/playframework/play-meta/blob/master/repos.md
      progress is being tracked here: https://github.com/playframework/play-meta/issues/24
	* [ ] PR+tag builds only https://github.com/playframework/playframework/pull/9139
	* [ ] PR+tag builds only https://github.com/lagom/lagom/pull/1841
* [x] Fix ssl-config situation https://github.com/playframework/play-meta/issues/4 @dwijnand

### Iceland

* [x] Dependency injection for Akka Typed in Play @marcospereira
* [ ] Follow the OpenShift Lagom guide @marcospereira
* [ ] Evaluate the impacts of Akka 2.6 in Play & Lagom @renatocaval
* [ ] Akka Projections @renatocaval & @marcospereira
    * [ ] Cluster distribution
    * [x] Output is a presentation to Play/Lagom team (invite Akka team as well)
* [ ] JUnit 5 support

**Comment by SethTisue** [comment]
<tiny voice> would https://github.com/lightbend/ssl-config/issues/142 fit?

**Comment by TimMoore** [comment]
The changes around the Lagom example applications seem worth discussing more widely, both within Lightbend and in the community.

**Comment by dwijnand** [comment]
> The changes around the Lagom example applications seem worth discussing more widely, both within Lightbend and in the community.

@TimMoore let's discuss that in https://github.com/playframework/play-meta/issues/33.  Both Ignasi and Renato hold some different opinions so let's agree on a plan there first.  Renato has agreed to lay out his ideas there so Ignasi and others can discuss off of those.

**Comment by dwijnand** [comment]
> <tiny voice> would [lightbend/ssl-config#142](https://github.com/lightbend/ssl-config/issues/142) fit?

@SethTisue I don't have the background knowledge in ssl-config and play to efficiently push forward on that change.  Last time I spoke about it with @marcospereira he said he'd try and review Will's various PRs.  The best I can do is restore ssl-config's master so it builds with play master, but that would mean it wouldn't pass on Java 11 anymore, meaning the community build would need to re-fork (or add the change its existing fork, if present).  Is that result a net-win for you or neutral/worst?

**Comment by SethTisue** [comment]
It's more important that we not let Lightbend stuff lag too far behind in the JDK 8 build.
I'm not super concerned about building on JDK 11, it's only a nice-to-have.
I can't recall if there is anything here that affects *running* published artifacts on JDK 11. If so, that's a potential issue for users and customers and probably means it should be a priority after all (not urgently, but not many months of back-burnering, either?).

**Comment by dwijnand** [comment]
I agree with the priority of not letting ssl-config -> play -> lagom lag, due to being forked in the community build.  So I'll unbreak the situation by reverting the initial change.

I also agree with the less-important-but-still-important goal of getting it working on Java 11 (it's one of the current priorities: "Drop allow_failures for AdoptOpenJDK 11" above).

**Comment by TimMoore** [comment]
> @TimMoore let's discuss that in #33.

Sounds good, but I still suggest making this issue/discussion more visible. I don't think all interested parties will necessarily be watching this repo.
